src/components/new-safe/create/steps/StatusStep/index.tsx

```diff
@@ -39,7 +38,12 @@ export const CreateSafeStatus = ({ data, setProgressColor }: StepRenderProps<New
   const isWrongChain = useIsWrongChain()
   const isConnected = wallet && !isWrongChain
 
-  const { handleCreateSafe } = useSafeCreation(pendingSafe, setPendingSafe, status, setStatus, data.willRelay)
+  // The willRelay flag can come from the previous step or from local storage
+  const willRelay = data.willRelay || pendingSafe?.willRelay
+  const initialStatus = willRelay ? SafeCreationStatus.PROCESSING : SafeCreationStatus.AWAITING
```

Comment:
The initial status is also referenced in `useSafeCreation` on line 146-147. What do you think about creating a helper for this? e.g.

```ts
const getInitialCreationStatus = (willRelay: boolean) => willRelay ? SafeCreationStatus.PROCESSING : SafeCreationStatus.AWAITING
```